### PR TITLE
06313 - AddressBook Config Text Supports Noncontiguous NodeId

### LINKED ISSUE #5893: remove `isOwnHost()` method from Address.java
The `isOwnHost()` property is only being used by the browser to launch local nodes.   The property is set to false when deserializing addresses.  The field and methods should be removed.  There should be a way for the browser to determine which addresses to launch without needing to add the marker to the addresses themselves.

### LINKED ISSUE #6313: Noncontiguous NodeId - Update AddressBook serialization/deserialization
* Replace the long id field with NodeId field in binary serialization
* Update text based address book representations to include node id
* Update or create any related unit tests with respect to noncontiguous NodeId

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
* The text serialization of AddressBook has been updated with NodeIds.   The addresses must be in ascending order.  
* The `isOwnHost()` method has been removed from Address.java

Fixes #6313 
Fixes #5893 

IMPACTS:
* JRS TESTS 
* * ALL JRS TESTS WILL FAIL. 
* * All config.txt for JRS tests will need to be updated with consecutive node ids in the addresses.  
* * * See the config.txt being checked in with this PR. 
* DEV-OPS / Hedera
* * The config.txt address book needs to be updated with the new field for each address. 
* * If the nickname is already the numeric node id, then add a new nickname for each address.

### CODE REVIEW

**Comment by poulok** [comment]
Please be sure to discuss the impact of this change with @dalvizu on his team and ensure necessary changes can be made on his side at the appropriate time before merging.

**Review by JeffreyDallas — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Review by jeromy-cannon — DISMISSED** [review]

**Comment by JeffreyDallas** [comment]
Since regression test flow is shared by both platform tests and service/hedera tests,
if just simply updating config.txt in regression repo, then tests of hedera would fail since they 
are still using old platform code with old config.txt format.

Please hold on the merge let me make necessary plan/changes to support the transition.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java:161]:
```suggestion
        final boolean fromIsLocal = AddressBookNetworkUtils.isLocal(from);
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java:162]:
```suggestion
        final boolean toIsLocal = AddressBookNetworkUtils.isLocal(to);
```

**Review by hendrikebbers — DISMISSED** [review]

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/internal/ApplicationDefinition.java
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/internal/ApplicationDefinition.java:44]: `@NonNull`

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:42]: `@NonNull` for return type

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:406]: 🥳

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/legacy/LegacyConfigProperties.java
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/legacy/LegacyConfigProperties.java:59]: Please add javadocs to these methods.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:37]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:45]: changing the exception throw to returning false.  The effect is that the address will not start locally, but any nodes that are local will be started.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:63]: Agreed, changed it to your recommendation.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:120]: No necessary reason in this case.  There was another unit test where the size of the address book was scaling in a for loop and I think I just put it inside the loop by muscle memory from working the other unit test.   I moved it outside the for loop.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/internal/ApplicationDefinition.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/internal/ApplicationDefinition.java:44]: adding marker and check.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:42]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/legacy/LegacyConfigProperties.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/legacy/LegacyConfigProperties.java:59]: added

File: platform-sdk/swirlds-platform-core/src/test/resources/com/swirlds/platform/config/legacy/config.txt
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/test/resources/com/swirlds/platform/config/legacy/config.txt]: I think that makes sense. Having a few extra valid lines is nice to quickly uncomment

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:45]: The illegal state exception isn't thrown anymore

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java]: My mind is still on vacation.  :-) removed.

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/CryptoStatic.java
Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/CryptoStatic.java:364]: I think this would probably be a critical failure at startup, but I'm not 100% sure.

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/sdk/settings.txt
Comment (edward-swirldslabs) [reply platform-sdk/sdk/settings.txt:19]: added newline

**Review by cody-littley — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by alittley — DISMISSED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM as per Services.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:53]: No idea about the internal of `TextTable` but it looks to me that `TextTable` should add the "," instead of doing it for each column by hand.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:78]: do we have the `addressBookText` defined somewhere. Otherwise I would like to see a sentence in the javadoc regarding the `addressBookText.split("\\r?\\n")` that shows a simple sample

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:108]: ok, from my point of view it really makes sense to define how `addressText` is structured. Otherwise you always need to do reverse eng to understand the code ;)

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:632]: Should `AddressBook` implement `Iterable<NodeId>`?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java:161]: Why is `isLocal` not a method of `Address`?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:57]: Is that really correct or should an exception be thrown?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:69]: null annotation missing

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:53]: I agree, Textable should be configurable with whether or not to add the comma.   I'll look into how hard it would be to add it.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:78]: It's the format in config.txt.  It is described well there.  The split you're quoting splits on newlines between addresses.  The split handles both windows and the rest of the world formats for new lines.   I will add more comments to the javadoc.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:108]: I will add more comments.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:632]: It implements `Iterable<Address>` which contains the node Ids.  Now that NodeId is  non-contiguous, the concept of a NodeIndex in the range of [0 .. (N-1)] has become relevant in contrast.  The Node Index is used in places that relied on the NodeIds being contiguous starting from 0.  In the Browser, the GUI is tied to using logic based off of index for contiguous display of windows.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/OutboundConnectionCreator.java:161]: Because `Address` and `AddressBook` are serialized as part of the state and the association between an address and an executing node is logic outside the address/addressbook itself.    Everyone gets the same state and addressbook.  But not everyone is going to have the same result for "isLocal".   Divergence in a synchronized data structure is not good design.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:53]: I created an issue for this as it is not part of the scope of this PR: https://github.com/hashgraph/hedera-services/issues/7267

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:78]: I have added a javadoc comment at the top of the `AddressBookUtils.java` that contains the parsing and serialization methods.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java:632]: no change made.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookNetworkUtils.java:57]: Initially I had both, an error in the log and an exception.   Someone made a comment asking why we need both.  I deleted the exception instead of the log message.    This method is replacing `isOwnHost` and is called where where `address.isOwnHost()` would be called and returning the same value, a boolean false.  I noticed that Exceptions seem to be dumped and not logged by executors.

Now throwing an IllegalStateException

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:33]: I think it would be more helpful to show the format something like this

`address,<nodeId>,<nickname>...`

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:38]: Perhaps you could rewrite this in the following form:

```
Each line in the config.txt address book contains the following elements, separated by commas. Each address line is terminated by a newline ("\n" or "\r\n") or end of file.

<ul>
<li>the keyword "address"</li>
...
<li>a memo (optional)</li>
</ul>
```

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:33]: I am also ok with this strategy

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:38]: will do

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:33]: makes sense.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:33]: It looks like there is a fragment of the old javadoc hanging around at the end of this line.
```suggestion
 * Each line in the config.txt address book contains the following comma separated elements:
```

Comment (alittley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:33]:
```
config.txt address book: keywork "address" followed by a comma, followed by the node ID, followed by a
```

This part of line 33 needs to be deleted, it doesn't seem to belong

Some sort of mistake was made changing this doc - this line is mangled

**Review by jeromy-cannon — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:104]: Why do you always do 2 spaces between sentences?

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:104]: I am 43 and learned to type pre-internet.  The 2 spaces is automatic.  When text was typed on type writers, it was the standard to add two spaces after periods.  HTML rendering eliminated the double spacing and collapsed all white space.  This caused the standard to change.  I just haven't adapted since.

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:104]: https://www.theatlantic.com/science/archive/2018/05/two-spaces-after-a-period/559304/

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java:104]: https://www.masterclass.com/articles/how-many-spaces-after-a-period

**Pull request merged**
